sim_pkg/__main__.py

```diff
@@ -6,85 +8,37 @@
 import socketserver
 import csv
 import numpy as np
-from init_pose import init
 import argparse
+from sim_pkg.config import Configuration
 
-# sim_process_ = 0 
 robot_processes_ = []
-# vis_processes_ = 0
 parser = argparse.ArgumentParser(description='Run the simulator')
 
-parser.add_argument('-fn', '--filename', type=str, help="Name of the file without the .py extension")
+parser.add_argument('-fn', '--filename', type=str,
+                    help="Name of the file without the .py extension")
 
 args = parser.parse_args()
 
-def write_to_csv(free_port):
-    """
-    Write avaliable port number to csv file
-    """
-    f = open('port.csv', 'w')
-    num = [free_port]
-    with f:
-        # create the csv writer
-        writer = csv.writer(f)
-
-        # write a row to the csv file
-        writer.writerow(num)
-
-    # close the file
-    f.close()
-
-def init_pos(swarmsize):
```

Comment:
I do not see the purpose in this. If, for example, I, as a user, want to configure the initial positions, then let me do so in `init.csv`. If not, then why are you populating this with some defaults? Just fetch those defaults straight from memory as you did in `simulator.py`.

I strongly prefer:
https://github.com/devbhura/swarm_simulation/pull/14/files#diff-f899b96f09a0c5456b6d7a4e99c701de3b0fde75529142b58f4f4eb4e8c72a70R58-R78